recipes/proposal/all/conanfile.py

```diff
@@ -50,9 +50,14 @@ def requirements(self):
 
     @property
     def _minimum_compilers_version(self):
-        return {"gcc": "5", "clang": "5", "apple-clang": "5"}
+        return {"Visual Studio": "15", "gcc": "5", "clang": "5", "apple-clang": "5"}
 
     def validate(self):
+
+        if (self.settings.os == "Windows") and self.options.shared:
+            raise ConanInvalidConfiguration(
+                "Can not build shared library on Windows."
+            )
```

Comment:
```suggestion
        if self.settings.compiler == "Visual Studio" and self.options.shared:
            raise ConanInvalidConfiguration(
                "Can not build shared library on Visual Studio."
            )
```

No, it works with MinGW by default. exporting symbols or not is not a "windows" concept.